src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionListProvider.cs

```diff
@@ -53,17 +51,15 @@ internal class DelegatedCompletionListProvider : CompletionListProvider
             VSInternalClientCapabilities clientCapabilities,
             CancellationToken cancellationToken)
         {
-            var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
-            var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
-            var projection = GetProjection(absoluteIndex, codeDocument, sourceText);
+            var projection = await documentContext.GetProjectionAsync(absoluteIndex, _documentMappingService, cancellationToken).ConfigureAwait(false);
```

Comment:
> Does `GetCompletionProjectionAsync` make sense in context?

The projection isn't specific to completion, and in fact in this PR i'm genericising it so it can be reused by everything.